src/mjolnir/adminbuilder.cc

```diff
@@ -495,7 +495,7 @@ bool BuildAdminFromPBF(const boost::property_tree::ptree& pt,
   sql = "INSERT INTO admins (admin_level, iso_code, parent_admin, name, name_en, ";
   sql +=
       "drive_on_right, allow_intersection_names, default_language, supported_languages, geom) VALUES (?,?,?,?,?,?,?,?,?,";
-  sql += "CastToMulti(GeomFromText(?, 4326)))";
+  sql += "ST_MakeValid(CastToMulti(GeomFromText(?, 4326))))";
```

Comment:
This properly handles bowties and other self-intersections by splitting them into multiple rings where possible